hknweb/tutoring/views.py

```diff
@@ -21,5 +27,173 @@ def index(request):
+    return prefs
+
+# Generates file that will be fed into algorithm
+def prepare_algorithm_input(request):
+    input = {}
+    courses = []
+    for course in Course.objects.all():
+        courses.append(course.name)
+    input["courseName"] = courses
+    tutors = []
+    for tutor in Tutor.objects.all():
+        tutor_dict = {}
+        tutor_dict["tid"] = tutor.id
+        tutor_dict["name"] = tutor.name
+        slot_time_prefs = []
+        slot_office_prefs = []
+        slot_opposite_office_prefs = []
+        for pref in tutor.get_slot_preferences():
+            slot_time_prefs.append(pref.time_preference)
+            slot_office_prefs.append(pref.office_preference)
+            slot_opposite_office_prefs.append(-pref.office_preference)
+        tutor_dict["timeSlots"] = slot_time_prefs + slot_time_prefs
+        tutor_dict["officePrefs"] = slot_office_prefs + slot_opposite_office_prefs
+        course_prefs = []
+        for pref in tutor.get_course_preferences():
+            course_prefs.append(pref.preference)
+        tutor_dict["courses"] = course_prefs
+        tutor_dict["adjacentPref"] = tutor.adjacent_pref
+        tutor_dict["numAssignments"] = tutor.num_assignments
+        tutors.append(tutor_dict)
+    input["tutors"] = tutors
+    slots = []
+    cory_course_prefs = get_office_course_preferences(0)
+    soda_office_prefs = get_office_course_preferences(1)
+    for slot in Slot.objects.all().order_by('slot_id'):
+        slot_dict = {}
+        slot_dict["sid"] = slot.slot_id
+        slot_dict["name"] = "Slot {}".format(slot.slot_id)
+        slot_dict["adjacentSlotIDs"] = get_adjacent_slot_ids(slot.slot_id)
+        if slot.room == 0:
+            slot_dict["courses"] = cory_course_prefs
+        else:
+            slot_dict["courses"] = soda_office_prefs
+        slot_dict["day"] = slot.timeslot.get_day()
+        slot_dict["hour"] = slot.timeslot.hour
+        slot_dict["office"] = slot.get_office()
+        slots.append(slot_dict)
+    input["slots"] = slots
+    return JsonResponse(input)
+
+def get_adjacent_slot_ids(slot_id):
+    adjacent = []
+    if slot_id < 10:
+        adjacent.append(slot_id + 10)
+    elif slot_id >= 50:
+        adjacent.append(slot_id - 10)
+    else:
+        adjacent.append(slot_id + 10)
+        adjacent.append(slot_id + 10)
+    return adjacent
+
+#Take in json data and 
+def generate_schedule(request):
+    if request.method == 'POST':
+        form = TutoringAlgorithmOutputForm(request.POST, request.FILES)
+        if form.is_valid():
+            output = request.FILES['output']
+            data = json.loads(output.read().decode("utf-8"))
+            for slot_id in data:
+                slot = Slot.objects.get(slot_id=slot_id)
+                tutor_ids = data[slot_id]
+                for id in tutor_ids:
+                    tutor = Tutor.objects.get(id=id)
+                    slot.tutors.add(tutor)
+    return redirect('/tutoring/')
+    # if request.method == 'POST':
```

Comment:
Do we need these pieces of commented out code? If so, let's document them and put in an issue if applicable; otherwise, let's removed them?